cmd/ncproxy/service.go

```diff
@@ -0,0 +1,221 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+const serviceName = "ncproxy"
+
+var (
+	panicFile *os.File
+	oldStderr windows.Handle
+)
+
+type handler struct {
+	fromsvc  chan error
+	serveErr chan error
+}
+
+func getServicePath() (string, error) {
+	p, err := exec.LookPath(os.Args[0])
```

Comment:
Using `os.Executable` seems more stable here, since it will return the actual binary path, rather than the first command line argument.

Reply:
Sure, changing.

Reply:
Done